examples/grpc/src/server.rs

```diff
@@ -0,0 +1,52 @@
+use tonic::{transport::Server, Request, Response, Status};
+
+use hello_world::greeter_server::{Greeter, GreeterServer};
+
+use clap::Parser;
+use hello_world::{HelloReply, HelloRequest};
+
+pub mod hello_world {
+    tonic::include_proto!("helloworld"); // The string specified here must match the proto package name
+}
+
+#[derive(Parser)]
+struct Cli {
+    #[arg(default_value_t = String::from("[::0]"))]
+    address: String,
+    #[arg(short, long, default_value_t = String::from("50001"))]
+    port: String,
+}
+
+#[derive(Debug, Default)]
+pub struct MyGreeter {}
+
+#[tonic::async_trait]
+impl Greeter for MyGreeter {
+    async fn say_hello(
+        &self,
+        request: Request<HelloRequest>, // Accept request of type HelloRequest
+    ) -> Result<Response<HelloReply>, Status> {
+        // Return an instance of type HelloReply
+        println!("Got a request: {:?}", request);
+
+        let reply = hello_world::HelloReply {
+            message: format!("Hello {}!", request.into_inner().name).into(), // We must use .into_inner() as the fields of gRPC requests and responses are private
```

Comment:
The function mentioned in the comment doesn't match the function being used.

Reply:
`request.into_inner().name` uses it, right?

Reply:
Oh yeah I see it now